Code/GraphMol/ForceFieldHelpers/UFF/Builder.cpp

```diff
@@ -488,7 +488,7 @@ void addTorsions(const ROMol &mol, const AtomicParamVect &params,
 
   // find all of the torsion bonds:
   std::vector<MatchVectType> matchVect;
-  static const ROMol *defaultQuery = SmartsToMol(defaultTorsionBondSmarts);
+  const ROMol *defaultQuery = SmartsToMol(defaultTorsionBondSmarts);
```

Comment:
A little threading lesson here.  If build neighbor matrix is used in a threaded fashion static const in clang and g++ actually has a mutex placed around it in case buildNeighborMatrix is called multiple times in a thread, but in msvc there is no such protection so "static const" would not be thread safe and you would need to make your own mutex (boost has an initialize once functionality that we use elsewhere).

In the current case (replaces static const with const) case you are actually making the defaultQuery every time so this is now a leak.  Note that in file scope a const last for the programs lifetime, but for a function it does not (but static would here, if you made it thread safe).

Note the const boost::uint8_t above can be deduced at compile time, but the SmartsToMol cannot.

The safest thing here would be:

```
const ROMol *query = SmartsToMol(torsionBondSmarts);
```

and take the minor performance hit.  Otherwise we need to make a protected once only init for the default query.  Here is a modified thread-safe example from FunctionalGroupHierarchy.cpp:

```

#ifdef RDK_THREADSAFE_SSS
#include <boost/thread/once.hpp>
#endif

 
namespace {
const std::string defaultTorsionBondSmarts = "CCC";

RWMol & default_query_get() {
  // returns the query (needs to be initialized by default_query_create)
  static RWMol mol;
  return mol;
}

void default_query_create() {
  // this is only called once to make the query
  ROMol * query = SmartsToMol(defaultTorsionBondSmarts);
  default_query_get() = *query;
  delete query;
}
}

const ROMol & GetDefaultQuery() {
#ifdef RDK_THREADSAFE_SSS
#ifdef BOOST_THREAD_PROVIDES_ONCE_CXX11
  static boost::once_flag flag;
#else
  static boost::once_flag flag = BOOST_ONCE_INIT;
#endif
  boost::call_once(&default_query_create, flag);
#else
  static bool loaded = false;
  if (!loaded) {
    default_query_create();
    loaded = true;
  }
#endif
  return default_query_get();
}
```